src/components/DiagramContainer.tsx

```diff
@@ -93,28 +64,174 @@ class DiagramContainer extends React.Component<DiagramProps, DiagramState> {
     props: DiagramProps,
     state: DiagramState
   ) {
-    console.log("state received from Diagram Manager: ", props.dataFromApp);
-    if (state.cachedState != props.dataFromApp) {
-      console.log("updating state, with state received from Diagram Manager");
-      // this.refreshNodeIndex(props.dataFromApp);
-      return {
-        cachedState: props.dataFromApp,
-        nodeDataArray: props.dataFromApp,
-        skipsDiagramUpdate: true,
-      };
+    console.log("state received from DiagramManager: ", props.dataFromApp);
+    if (state.cachedState === props.dataFromApp) {
+      return null;
     }
+
+    console.log("updating state, with state received from Diagram Manager");
+    // this.refreshNodeIndex(props.dataFromApp);
+    return {
+      cachedState: props.dataFromApp,
+      nodeDataArray: props.dataFromApp,
+      skipsDiagramUpdate: true,
+    };
   }
 
   /**
    * Handle any relevant DiagramEvents, in this case just selection changes.
    * On ChangedSelection, find the corresponding data and set the selectedData state.
    * @param e a GoJS DiagramEvent
    */
-  public handleDiagramEvent(e: go.DiagramEvent) {
-    const name = e.name;
-    switch (name) {
-      default:
-        break;
+  public handleDiagramEvent(event: go.DiagramEvent) {
+    return;
+  }
+
+  private sendBackendUpdates(obj: go.IncrementalData) {
+    const insertedNodeKeys = obj.insertedNodeKeys;
+    const removedNodeKeys = obj.removedNodeKeys;
+    const modifiedNodeData = obj.modifiedNodeData;
+
+    const insertedLinkKeys = obj.insertedLinkKeys;
+    const removedLinkKeys = obj.removedLinkKeys;
+    const modifiedLinkData = obj.modifiedLinkData;
+
+    // node added or modified
+    if (modifiedNodeData) {
+      for (let nodeData of modifiedNodeData) {
+        if (insertedNodeKeys && insertedNodeKeys.includes(nodeData["key"])) {
```

Comment:
Ooo I like the way you used insertedNodeKeys to check if the modified node is a new node. We could use this logic in general for adding nodes instead of the provided way where we check both modifiedNodeData and insertedNodeData. Can be done in a different cleanup pr though